### ISSUE TITLE
Duplicate in train and test set

### ISSUE THREAD

**Issue by solitaire2015** [philipperemy/deep-learning-bitcoin#17]
Hi.
I find the code in data_generator.py line 64: `i = np.random.choice(n)`  ,`np.random.choice(n)` can produce same choices, it means there would be duplicate data in both training set and test set. 

And in line 65, `btc_slice = btc_df[i:i + slice_size]`, there would be some issues if the data slice is not continuous.
